### PR TITLE
fix(TDP-3579): lookup step in edition mode

### PR DESCRIPTION
**Pull request by amaalej**
**Link to the JIRA issue**
e.g. https://jira.talendforge.org/browse/TDP-3579

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
A random behaviour while editing creating a lookup step. Edition could lead to a step creation and vice-versa.

**(Optional) What is the new behavior?**
2 modes are set: edition mode & creation mode.

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/138/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/138/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/139/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/139/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/153/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/153/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/155/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/155/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/157/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/157/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/158/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/158/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/159/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/159/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/160/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/160/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/162/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/162/)

File: dataprep-webapp/src/app/components/lookup/lookup-controller.js [hunk 1]
```diff
@@ -31,84 +31,103 @@ export default function LookupCtrl($timeout, state, StateService,
...
-    /**
-     * @ngdoc method
-     * @methodOf data-prep.lookup.controller:LookupCtrl
-     * @name refreshLookupDatasetsSort
-     * @description refresh the actual sort parameter
-     * */
+	/**
+	 * @ngdoc method
+	 * @methodOf data-prep.lookup.controller:LookupCtrl
+	 * @name whereToLoadFrom
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/lookup/lookup-controller.js]: ok

File: dataprep-webapp/src/app/components/lookup/lookup-controller.js [hunk 2]
```diff
@@ -119,17 +138,17 @@ export default function LookupCtrl($timeout, state, StateService,
...
+	 * @name submit
+	 * @methodOf data-prep.lookup.controller:LookupCtrl
+	 * @description submits the lookup action
+	 */
 	vm.submit = function submit() {
 		EarlyPreviewService.deactivatePreview();
 		EarlyPreviewService.cancelPendingPreview();
 		let promise;
-		const lookupStep = vm.state.playground.lookup.step;
+		const lookupStep = state.playground.stepInEditionMode;
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/lookup/lookup-controller.js]: residue between ancient and new code, but ok

File: dataprep-webapp/src/app/components/lookup/lookup-controller.spec.js [hunk 3]
```diff
@@ -1,367 +1,413 @@
...
+			// given
+			const ctrl = createController();
+			stateMock.playground.stepInEditionMode = step;
+
+			// when
+			ctrl.whereToLoadFrom(dsActions[0]);
+
+			// then
+			expect(LookupService.fetchLookupDatasetContent)
+				.toHaveBeenCalledWith('lookup_dataset_id', dsActions[0])
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/lookup/lookup-controller.spec.js]: ok

File: dataprep-webapp/src/app/components/lookup/lookup-controller.spec.js [hunk 4]
```diff
@@ -1,367 +1,413 @@
...
+
+			// then
+			expect(LookupService.loadFromAction).toHaveBeenCalledWith(dsActions[0])
+		}));
+	});
+
+	describe('preview', () => {
+		it('should trigger lookup preview', inject((EarlyPreviewService) => {
+			//given
+			var ctrl = createController();
```
Comment (amaalej) [inline dataprep-webapp/src/app/components/lookup/lookup-controller.spec.js]: ok

File: dataprep-webapp/src/app/services/lookup/lookup-service.js [hunk 5]
```diff
@@ -53,7 +56,7 @@ export default class LookupService {
 							return;
 						}
 
-						const step = this._getSelectedColumnLastLookup();
+						const step = this.state.playground.stepInEditionMode;
```
Comment (amaalej) [inline dataprep-webapp/src/app/services/lookup/lookup-service.js]: ok

File: dataprep-webapp/src/app/services/state/lookup/lookup-state-service.js [hunk 6]
```diff
@@ -184,14 +183,28 @@ export function LookupStateService() {
...
 	 * @ngdoc method
+	 * @name _getDsId
+	 * @methodOf data-prep.services.state.service:LookupStateService
+	 * @private
+	 * @param {object} lookupDataset lookup dataset
+	 * @return {string} lookup dataset id
+	 * @description extracts the lookup dataset id from the parameters
+	 */
+	function _getDsId(lookup) {
+		return lookup.parameters.find(param => param.name === 'lookup_ds_id').default;
```
Comment (amaalej) [inline dataprep-webapp/src/app/services/state/lookup/lookup-state-service.js]: It comes from backend, so backend issue

File: dataprep-webapp/src/app/services/state/lookup/lookup-state-service.js [hunk 7]
```diff
@@ -184,14 +183,28 @@ export function LookupStateService() {
...
 	 * @param {object} data The data
 	 * @description Create the checkboxes definition for each column
 	 */
-	function createColumnsCheckboxes(data) {
-		const addedColIds = lookupState.step ?
-			_.map(lookupState.step.actionParameters.parameters.lookup_selected_cols, 'id') :
+	function createColumnsCheckboxes(data, currentStep) {
+		const addedColIds = currentStep &&
+			currentStep.actionParameters.parameters.lookup_ds_id === _getDsId(lookupState.dataset) ?
+			_.map(currentStep.actionParameters.parameters.lookup_selected_cols, 'id') :
```
Comment (amaalej) [inline dataprep-webapp/src/app/services/state/lookup/lookup-state-service.js]: ok

File: dataprep-webapp/src/app/services/state/playground/playground-state-service.spec.js [hunk 8]
```diff
@@ -171,6 +171,18 @@ describe('Playground state service', () => {
             expect(playgroundState.grid.selectedColumns).toEqual([{ id: '0000'}]);
             expect(LookupStateService.setVisibility).toHaveBeenCalledWith(true);
         }));
+
+        it('should set set Lookup Visibility flag', inject((playgroundState, LookupStateService, PlaygroundStateService) => {
```
Comment (amaalej) [inline dataprep-webapp/src/app/services/state/playground/playground-state-service.spec.js]: ok

File: dataprep-webapp/src/app/services/state/lookup/lookup-state-service.spec.js [hunk 9]
```diff
@@ -1,471 +1,342 @@
...
-            //then
-            expect(lookupState.datasets[0].model.created).toBe('a');
-            expect(lookupState.datasets[1].model.created).toBe('b');
-            expect(lookupState.datasets[2].model.created).toBe('C');
-        }));
-    });
+describe('Lookup state service', () => {
+	'use strict';
+
+	var data = {
```
Comment (amaalej) [inline dataprep-webapp/src/app/services/state/lookup/lookup-state-service.spec.js]: ok

File: dataprep-webapp/src/app/services/state/lookup/lookup-state-service.spec.js [hunk 10]
```diff
@@ -1,471 +1,342 @@
...
+			{ tdpId: 6, firstname: 'Papa' },
+			{ tdpId: 7, firstname: 'Pepe' },
+			{ tdpId: 8, firstname: 'Pipi' },
+			{ tdpId: 9, firstname: 'Popo' },
+			{ tdpId: 10, firstname: 'Pupu' },
+			{ tdpId: 11, firstname: 'Pypy' },
+		],
+	};
+
+	var initialColumnCheckboxes = [
```
Comment (amaalej) [inline dataprep-webapp/src/app/services/state/lookup/lookup-state-service.spec.js]: ok

File: dataprep-webapp/src/app/services/state/lookup/lookup-state-service.spec.js [hunk 11]
```diff
@@ -1,471 +1,342 @@
...
+				},
+				{
+					name: 'lookup_selected_cols',
+					type: 'list',
+					default: '',
+				},
+			],
+		},
+	];
+	var lookupAction = actions[0];
```
Comment (amaalej) [inline dataprep-webapp/src/app/services/state/lookup/lookup-state-service.spec.js]: ok

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/167/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/167/)

**Pull request merged**